### ISSUE TITLE
Report a common mistake of into first argument as transducers(aka functions)

### ISSUE THREAD

**Issue by souenzzo** [clj-kondo/clj-kondo#1889]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**

Kondo currently warn us when we use a wrong value in as the first argument of `into`

```shell
echo '(into "" cat {:a 42})' | clj-kondo --lint -
<stdin>:1:7: error: Expected: collection, received: string.
linting took 72ms, errors: 1, warnings: 0
```

In my experience, it is common to forgot the first argument, like this:

```shell
echo '(into cat {:a 42})' | clj-kondo --lint -
linting took 60ms, errors: 0, warnings: 0
```

In this case, it does not report as something wrong

**Describe the solution you'd like**

It would be nice if it report if the first argument is a function

```shell
echo '(into cat {:a 42})' | clj-kondo --lint -
<stdin>:1:7: error: Expected: collection, received: function.
linting took 78ms, errors: 1, warnings: 0
```

At least, a few common cases: `into (map ...`, `into (comp ...`, ``into cat ...``

**Comment by borkdude** [comment]
This already works with clj-kondo on master:

```
$ echo '(into cat {:a 42})' | clj-kondo --lint -
<stdin>:1:7: error: Expected: collection, received: function.
```
